plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -82,190 +82,414 @@ function executeCLI($scriptName, &$args) {
+
+			if (!$authorGroup) {
+				$invalidRowsFile->fputcsv(array_merge($fieldsList, [__('plugins.importexport.csv.noAuthorGroup', ['press' => $data->pressPath])]));
+				continue;
+			}
+
+			$filePath = "{$basePath}/{$data->filename}";
+			if (!is_readable($filePath)) {
+				$invalidRowsFile->fputcsv(array_merge($fieldsList, [__('plugins.importexport.csv.invalidAssetFilename', ['title' => $data->title])]));
+				continue;
+			}
+
+			$coverImgExtension = pathinfo(mb_strtolower($data->bookCoverImage), PATHINFO_EXTENSION);
+
+			if (!in_array($coverImgExtension, $allowedFileTypes)) {
+				$invalidRowsFile->fputcsv(array_merge($fieldsList, [__('plugins.importexport.common.error.invalidFileExtension')]));
+				continue;
+			}
+
+			if ($data->bookCoverImage) {
+				$srcFilePath = "{$basePath}/{$data->bookCoverImage}";
+
+				if (!is_readable($srcFilePath)) {
+					$invalidRowsFile->fputcsv(array_merge($fieldsList, [__('plugins.importexport.csv.invalidCoverImageFilename', ['title' => $data->title])]));
+					continue;
+				}
+			}
+
+			$categoriesList = explode(';', $data->categories);
+			$dbCategories = [];
+
+			foreach($categoriesList as $categoryTitle) {
+				$dbCategory = $cachedCategories["{$categoryTitle}_{$press->getId()}"]
+					?? $cachedCategories["{$categoryTitle}_{$press->getId()}"] = $categoryDao->getByTitle(trim($categoryTitle), $press->getId(), $data->locale);
 
-		if (is_array($data) && count($data) > 0) {
+				if (!is_null($dbCategory)) {
+					$dbCategories[] = $dbCategory;
+				}
+			}
+
+			if (count($categoriesList) < count($dbCategories)) {
+				$invalidRowsFile->fputcsv(array_merge($fieldsList, [__('plugins.importexport.csv.allCategoriesMustExists')]));
+				continue;
+			}
 
-			$userDao = DAORegistry::getDAO('UserDAO'); /* @var $userDao UserDAO */
-			$user = $userDao->getByUsername($username);
-			if (!$user) {
-				echo __('plugins.importexport.csv.unknownUser', array('username' => $username)) . "\n";
-				exit();
+			// All requirements passed. Start processing from here.
+			$extension = $fileManager->parseFileExtension($data->filename);
+
+			$submission = $submissionDao->newDataObject();
+			$submission->setContextId($press->getId());
+			$submission->stampLastActivity();
+			$submission->stampModified();
+			$submission->setStatus(STATUS_PUBLISHED);
+			$submission->setWorkType($data->isEditedVolume == 1 ? WORK_TYPE_EDITED_VOLUME : WORK_TYPE_AUTHORED_WORK);
+			$submission->setCopyrightNotice($press->getLocalizedSetting('copyrightNotice'), $data->locale);
+			$submission->setLocale($data->locale);
+			$submission->setStageId(WORKFLOW_STAGE_ID_PRODUCTION);
+			$submission->setSubmissionProgress(0);
+			$submission->setData('abstract', $data->abstract, $data->locale);
+
+			$pressSeries = $series["{$data->seriesPath}_{$press->getId()}"]
+				?? $series["{$data->seriesPath}_{$press->getId()}"] = $data->seriesPath
+					? $seriesDao->getByPath($data->seriesPath, $press->getId())
+					: null;
```

Comment:
This will not work as intended, see the output of `echo 4 ?? 1 ? 2 : 3`

```suggestion
			$pressSeries = $series["{$data->seriesPath}_{$press->getId()}"]
				?? ($series["{$data->seriesPath}_{$press->getId()}"] = $data->seriesPath
					? $seriesDao->getByPath($data->seriesPath, $press->getId())
					: null);
```